### DISCUSSION TITLE
Is there a way to re-execute only the failed test cases in serenity after the jenkins build gets completed

### DISCUSSION THREAD

**Discussion by robintyagi12** [serenity-bdd/serenity-core#2657] [General]
Hi Team,

We are executing serenity test cases through Jenkins and want to re-run only the failed test cases post Jenkins build completion, is there a way to capture the failed test cases information in some json or xml and refer that file to re-run only those failed test cases on Jenkins.

Please share your inputs, it would be really helpful. @wakaleo & @globalworming

**Comment by globalworming** [comment]
assuming maven https://maven.apache.org/surefire/maven-surefire-plugin/examples/rerun-failing-tests.html
```
<configuration>
    <rerunFailingTestsCount>2</rerunFailingTestsCount>
</configuration>
```
it's the same for the failsafe plugin

also try out the https://plugins.jenkins.io/flaky-test-handler/

also also: don't retry ever :wink: fix your system so you don't have to retry

**Reply by wakaleo** [reply]
What Andreas said. Especially the last bit.
